### ISSUE TITLE
can't build la-exprfilter.lo on linux and missing generic_avx2.cpp

### ISSUE THREAD

**Issue by pingplug** [vapoursynth/vapoursynth#492]
distro: Arch Linux, GCC 9.1.0
version: latest source in master branch for now (24e4d52e1580f2e8e229c9eb7a0838ff241e4f9a)
```
  CXX      src/core/libexprfilter_la-exprfilter.lo
../vapoursynth/src/core/exprfilter.cpp: In member function 'void {anonymous}::ExprCompiler::addInstruction(const {anonymous}::ExprInstruction&)':
../vapoursynth/src/core/exprfilter.cpp:198:16: warning: enumeration value 'MUX' not handled in switch [-Wswitch]
  198 |         switch (insn.op.type) {
      |                ^
../vapoursynth/src/core/exprfilter.cpp:198:16: warning: enumeration value 'DUP' not handled in switch [-Wswitch]
../vapoursynth/src/core/exprfilter.cpp:198:16: warning: enumeration value 'SWAP' not handled in switch [-Wswitch]
../vapoursynth/src/core/exprfilter.cpp: At global scope:
../vapoursynth/src/core/exprfilter.cpp:235:19: error: using typedef-name '{anonymous}::ExprCompiler128::jit' after 'struct'
  235 |     friend struct jit;
      |                   ^~~
../vapoursynth/src/core/exprfilter.cpp:234:92: note: '{anonymous}::ExprCompiler128::jit' has a previous declaration here
  234 |     typedef jitasm::function<void, ExprCompiler128, uint8_t *, const intptr_t *, intptr_t> jit;
      |                                                                                            ^~~
../vapoursynth/src/core/exprfilter.cpp:235:5: error: friend declaration does not name a class or function
  235 |     friend struct jit;
      |     ^~~~~~
../vapoursynth/src/core/exprfilter.cpp: In member function 'virtual void (* {anonymous}::ExprCompiler128::getCode())(void*, intptr_t*, intptr_t)':
../vapoursynth/src/core/exprfilter.cpp:1003:39: error: 'ExprObj' was not declared in this scope; did you mean 'ExprOp'?
 1003 |             void *ptr = mmap(nullptr, ExprObj.GetCodeSize(), PROT_READ | PROT_WRITE | PROT_EXEC, MAP_ANON | MAP_PRIVATE, 0, 0);
      |                                       ^~~~~~~
      |                                       ExprOp
../vapoursynth/src/core/exprfilter.cpp: At global scope:
../vapoursynth/src/core/exprfilter.cpp:1019:19: error: using typedef-name '{anonymous}::ExprCompiler256::jit' after 'struct'
 1019 |     friend struct jit;
      |                   ^~~
../vapoursynth/src/core/exprfilter.cpp:1018:92: note: '{anonymous}::ExprCompiler256::jit' has a previous declaration here
 1018 |     typedef jitasm::function<void, ExprCompiler256, uint8_t *, const intptr_t *, intptr_t> jit;
      |                                                                                            ^~~
../vapoursynth/src/core/exprfilter.cpp:1019:5: error: friend declaration does not name a class or function
 1019 |     friend struct jit;
      |     ^~~~~~
../vapoursynth/src/core/exprfilter.cpp: In member function 'virtual void (* {anonymous}::ExprCompiler256::getCode())(void*, intptr_t*, intptr_t)':
../vapoursynth/src/core/exprfilter.cpp:1578:39: error: 'ExprObj' was not declared in this scope; did you mean 'ExprOp'?
 1578 |             void *ptr = mmap(nullptr, ExprObj.GetCodeSize(), PROT_READ | PROT_WRITE | PROT_EXEC, MAP_ANON | MAP_PRIVATE, 0, 0);
      |                                       ^~~~~~~
      |                                       ExprOp
../vapoursynth/src/core/exprfilter.cpp: In function 'std::unique_ptr<{anonymous}::ExprCompiler> {anonymous}::make_compiler(int, int)':
../vapoursynth/src/core/exprfilter.cpp:1591:21: error: 'make_unique' is not a member of 'std'
 1591 |         return std::make_unique<ExprCompiler256>(numInputs);
      |                     ^~~~~~~~~~~
../vapoursynth/src/core/exprfilter.cpp:45:1: note: 'std::make_unique' is defined in header '<memory>'; did you forget to '#include <memory>'?
   44 | #include "jitasm.h"
  +++ |+#include <memory>
   45 | #ifndef VS_TARGET_OS_WINDOWS
../vapoursynth/src/core/exprfilter.cpp:1591:48: error: expected primary-expression before '>' token
 1591 |         return std::make_unique<ExprCompiler256>(numInputs);
      |                                                ^
../vapoursynth/src/core/exprfilter.cpp:1593:21: error: 'make_unique' is not a member of 'std'
 1593 |         return std::make_unique<ExprCompiler128>(numInputs);
      |                     ^~~~~~~~~~~
../vapoursynth/src/core/exprfilter.cpp:1593:21: note: 'std::make_unique' is defined in header '<memory>'; did you forget to '#include <memory>'?
../vapoursynth/src/core/exprfilter.cpp:1593:48: error: expected primary-expression before '>' token
 1593 |         return std::make_unique<ExprCompiler128>(numInputs);
      |                                                ^
../vapoursynth/src/core/exprfilter.cpp: In member function 'void {anonymous}::ExprInterpreter::eval(const uint8_t* const*, uint8_t*, int)':
../vapoursynth/src/core/exprfilter.cpp:1716:61: warning: suggest parentheses around comparison in operand of '!=' [-Wparentheses]
 1716 |                 registers[insn.dst] = (registers[insn.src1] > 0.0f != registers[insn.src2] > 0.0f) ? 1.0f : 0.0f;
../vapoursynth/src/core/exprfilter.cpp:1616:20: warning: enumeration value 'MEM_STORE_F16' not handled in switch [-Wswitch]
 1616 |             switch (insn.op.type) {
      |                    ^
../vapoursynth/src/core/exprfilter.cpp:1616:20: warning: enumeration value 'MUX' not handled in switch [-Wswitch]
../vapoursynth/src/core/exprfilter.cpp:1616:20: warning: enumeration value 'DUP' not handled in switch [-Wswitch]
../vapoursynth/src/core/exprfilter.cpp:1616:20: warning: enumeration value 'SWAP' not handled in switch [-Wswitch]
../vapoursynth/src/core/exprfilter.cpp: In function 'float {anonymous}::evalConstantExpr(const {anonymous}::ExpressionTreeNode&)':
../vapoursynth/src/core/exprfilter.cpp:2096:63: warning: suggest parentheses around comparison in operand of '!=' [-Wparentheses]
 2096 |     case ExprOpType::XOR: return evalConstantExpr(*node.left) > 0.0f != evalConstantExpr(*node.right) > 0.0f ? 1.0f : 0.0f;
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
../vapoursynth/src/core/exprfilter.cpp: In member function 'void {anonymous}::ExponentMap::combineConstants(const ValueIndex&)':
../vapoursynth/src/core/exprfilter.cpp:2225:31: error: 'powf' is not a member of 'std'; did you mean 'pow'?
 2225 |                 coeff *= std::powf(node->op.imm.f, it->second);
      |                               ^~~~
      |                               pow
../vapoursynth/src/core/exprfilter.cpp: In lambda function:
../vapoursynth/src/core/exprfilter.cpp:2321:77: error: could not convert 'memOpCodes' from 'initializer_list<const {anonymous}::ExprOpType>' to 'initializer_list<{anonymous}::ExprOpType>'
 2321 |             int lhsCategory = isConstant(*lhsNode) ? 2 : isOpCode(*lhsNode, memOpCodes) ? 1 : 0;
      |                                                                             ^~~~~~~~~~
      |                                                                             |
      |                                                                             initializer_list<const {anonymous}::ExprOpType>
../vapoursynth/src/core/exprfilter.cpp:2322:77: error: could not convert 'memOpCodes' from 'initializer_list<const {anonymous}::ExprOpType>' to 'initializer_list<{anonymous}::ExprOpType>'
 2322 |             int rhsCategory = isConstant(*rhsNode) ? 2 : isOpCode(*rhsNode, memOpCodes) ? 1 : 0;
      |                                                                             ^~~~~~~~~~
      |                                                                             |
      |                                                                             initializer_list<const {anonymous}::ExprOpType>
make: *** [Makefile:1629: src/core/libexprfilter_la-exprfilter.lo] Error 1
```
and another error
```
make: *** No rule to make target 'src/core/kerne/x86/generic_avx2.cpp', needed by 'src/core/kerne/x86/libvapoursynth_avx2_la-generic_avx2.lo'.  Stop.
make: *** Waiting for unfinished jobs....
```

**Comment by myrsloik** [comment]
Should work now

**Comment by pingplug** [comment]
another issue
```
  CXX      src/core/libexprfilter_la-exprfilter.lo
../vapoursynth/src/core/exprfilter.cpp: In member function 'void {anonymous}::ExprCompiler::addInstruction(const {anonymous}::ExprInstruction&)':
../vapoursynth/src/core/exprfilter.cpp:198:16: warning: enumeration value 'MUX' not handled in switch [-Wswitch]
  198 |         switch (insn.op.type) {
      |                ^
../vapoursynth/src/core/exprfilter.cpp:198:16: warning: enumeration value 'DUP' not handled in switch [-Wswitch]
../vapoursynth/src/core/exprfilter.cpp:198:16: warning: enumeration value 'SWAP' not handled in switch [-Wswitch]
../vapoursynth/src/core/exprfilter.cpp: In member function 'virtual void (* {anonymous}::ExprCompiler128::getCode())(void*, intptr_t*, intptr_t)':
../vapoursynth/src/core/exprfilter.cpp:1003:39: error: 'ExprObj' was not declared in this scope; did you mean 'ExprOp'?
 1003 |             void *ptr = mmap(nullptr, ExprObj.GetCodeSize(), PROT_READ | PROT_WRITE | PROT_EXEC, MAP_ANON | MAP_PRIVATE, 0, 0);
      |                                       ^~~~~~~
      |                                       ExprOp
../vapoursynth/src/core/exprfilter.cpp: In member function 'virtual void (* {anonymous}::ExprCompiler256::getCode())(void*, intptr_t*, intptr_t)':
../vapoursynth/src/core/exprfilter.cpp:1578:39: error: 'ExprObj' was not declared in this scope; did you mean 'ExprOp'?
 1578 |             void *ptr = mmap(nullptr, ExprObj.GetCodeSize(), PROT_READ | PROT_WRITE | PROT_EXEC, MAP_ANON | MAP_PRIVATE, 0, 0);
      |                                       ^~~~~~~
      |                                       ExprOp
../vapoursynth/src/core/exprfilter.cpp: In member function 'void {anonymous}::ExprInterpreter::eval(const uint8_t* const*, uint8_t*, int)':
../vapoursynth/src/core/exprfilter.cpp:1716:61: warning: suggest parentheses around comparison in operand of '!=' [-Wparentheses]
 1716 |                 registers[insn.dst] = (registers[insn.src1] > 0.0f != registers[insn.src2] > 0.0f) ? 1.0f : 0.0f;
../vapoursynth/src/core/exprfilter.cpp:1616:20: warning: enumeration value 'MEM_STORE_F16' not handled in switch [-Wswitch]
 1616 |             switch (insn.op.type) {
      |                    ^
../vapoursynth/src/core/exprfilter.cpp:1616:20: warning: enumeration value 'MUX' not handled in switch [-Wswitch]
../vapoursynth/src/core/exprfilter.cpp:1616:20: warning: enumeration value 'DUP' not handled in switch [-Wswitch]
../vapoursynth/src/core/exprfilter.cpp:1616:20: warning: enumeration value 'SWAP' not handled in switch [-Wswitch]
../vapoursynth/src/core/exprfilter.cpp: In function 'float {anonymous}::evalConstantExpr(const {anonymous}::ExpressionTreeNode&)':
../vapoursynth/src/core/exprfilter.cpp:2096:63: warning: suggest parentheses around comparison in operand of '!=' [-Wparentheses]
 2096 |    case ExprOpType::XOR: return evalConstantExpr(*node.left) > 0.0f != evalConstantExpr(*node.right) > 0.0f ? 1.0f : 0.0f;
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~

../vapoursynth/src/core/exprfilter.cpp: In lambda function:
../vapoursynth/src/core/exprfilter.cpp:2321:77: error: could not convert 'memOpCodes' from 'initializer_list<const {anonymous}::ExprOpType>' to 'initializer_list<{anonymous}::ExprOpType>'
 2321 |  lhsCategory = isConstant(*lhsNode) ? 2 : isOpCode(*lhsNode, memOpCodes) ? 1 : 0;
      |                                                              ^~~~~~~~~~
      |                                                              |
      |                                                              initializer_list<const {anonymous}::ExprOpType>

../vapoursynth/src/core/exprfilter.cpp:2322:77: error: could not convert 'memOpCodes' from 'initializer_list<const {anonymous}::ExprOpType>' to 'initializer_list<{anonymous}::ExprOpType>'
 2322 |  rhsCategory = isConstant(*rhsNode) ? 2 : isOpCode(*rhsNode, memOpCodes) ? 1 : 0;
      |                                                              ^~~~~~~~~~
      |                                                              |
      |                                                              initializer_list<const {anonymous}::ExprOpType>

make: *** [Makefile:1621: src/core/libexprfilter_la-exprfilter.lo] Error 1
```

**Comment by sekrit-twc** [comment]
This is a GCC bug. To fix it, you must update to GCC 9.2.

../vapoursynth/src/core/exprfilter.cpp:2321:77: error: could not convert 'memOpCodes' from 'initializer_list<const {anonymous}::ExprOpType>' to 'initializer_list<{anonymous}::ExprOpType>'
